remove unnecessary imports so WebGPU can be thinner

deltakosh opened the pull request:
Fixes #15580

bjsplat commented:
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

deltakosh commented:
@sebavan  / @Popov72 / @RaananW : I removed the hard coded imports to Engine extensions and moved then back to the engine class. That way WebGPUEngine can have its own version and not the engine version + webgpu version.

@RaananW / @sebavan : I want to make sure this is not a problem for ThinEngine

deltakosh commented:
Because I'm a GIT illiterate I also included the content from https://github.com/BabylonJS/Babylon.js/pull/15572

bjsplat commented:
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

bjsplat commented:
Snapshot stored with reference name:
refs/pull/15579/merge

Test environment: 
https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15579/merge/index.html

To test a playground add it to the URL, for example:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15579/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15579/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15579/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15579/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15579/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15579/merge#BCU1XR#0

bjsplat commented:
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15579/merge/testResults/webgl2playwright/index.html

ryantrem commented on packages/tools/viewer-alpha/src/viewerFactory.ts:
```diff
@@ -1,5 +1,5 @@
 // eslint-disable-next-line import/no-internal-modules
-import { RGBDTextureTools, type AbstractEngine, type AbstractEngineOptions, type EngineOptions, type WebGPUEngineOptions } from "core/index";
+import { type AbstractEngine, type AbstractEngineOptions, type EngineOptions, type WebGPUEngineOptions } from "core/index";
```
I think for the build this needs to be:
```ts
import type { AbstractEngine, AbstractEngineOptions, EngineOptions, WebGPUEngineOptions } from "core/index";
```
I think that was my mistake.

RaananW commented:
Working on resolving the es6 build issue (filesize and chunks)

RaananW reviewed (approved):
I don't see it influencing thinengine, the only thing you added to abstractengine already exists there

bjsplat commented:
Visualization tests for WebGL 1 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15579/merge/testResults/webgl1/index.html

If tests were successful afterwards, this report might not be available anymore.
